Junos dwdm interface sensor support

bergroth opened the pull request:
four new sensors
    Chromatic Dispersion ps/nm
    Delay s (seconds)
    Q_factor (dB)
    preFEC_BER (ratio)

  reuse of
    SNR, dbm, temperature, current

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4f92076e-925a-47a9-89e1-4fdb561587fd) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/4f92076e-925a-47a9-89e1-4fdb561587fd) completed: **1 updated code elements**

murrant commented:
Thanks.
It seems like the sensors need to be a little more generic.
q-factor -> quality_factor or just quality
prefec_ber -> ber (for junos, you can just set the description to PreFEC BER)

murrant commented on includes/discovery/sensors/frequency/junos.inc.php:
```diff
@@ -0,0 +1,41 @@
+
+echo 'JunOS ';
+
+$multiplier = 1000000;
```
Instead of multiplying by 1000000 here, you should probably just store the value in MHz.
This also avoids having to change the sensor limits.

murrant replied:
Ignore this. :)  The value should be stored in Hz.  (Wireless sensors are stored in MHz)

murrant commented on includes/polling/functions.inc.php:
```diff
@@ -166,7 +166,8 @@ function record_sensor_data($device, $all_sensors)
 
         $rrd_name = get_sensor_rrd_name($device, $sensor);
-        $rrd_def = RrdDefinition::make()->addDataset('sensor', 'GAUGE', -20000, 20000);
+
```
You probably don't need this, see comment about frequency.

bergroth commented:
Storing Frequency sensor as MHz and not multiplying shows the limits as -3.0kHz
and 3.0kHz when it should be -+3.0GHz. Looks a bit odd to me. Are there any way
do set the unit (MHz, KHz ...) in the discovery function?


On 2017-11-14 15:27, Tony Murray wrote:
>
> *@murrant* commented on this pull request.
>
> --------------------------------------------------------------------------------
>
> In includes/discovery/sensors/frequency/junos.inc.php
> <https://github.com/librenms/librenms/pull/7714#discussion_r150846926>:
>
> > @@ -0,0 +1,41 @@
> +<?php
> +/*
> + * LibreNMS
> + *
> + * Copyright (c) 2016 Neil Lathwood <neil@lathwood.co.uk>
> + * This program is free software: you can redistribute it and/or modify it
> + * under the terms of the GNU General Public License as published by the
> + * Free Software Foundation, either version 3 of the License, or (at your
> + * option) any later version.  Please see LICENSE.txt at the top level of
> + * the source code distribution for details.
> + *tis*/
> +
> +echo 'JunOS ';
> +
> +$multiplier = 1000000;
>
> Instead of multiplying by 1000000 here, you should probably just store the
> value in MHz.
> This also avoids having to change the sensor limits.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/librenms/librenms/pull/7714#pullrequestreview-76456953>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASQkHvx8gtBjZL9uRvyWIsOHF3IYuRTuks5s2aNMgaJpZM4QbtIV>.
>

murrant commented on includes/polling/functions.inc.php:
```diff
@@ -166,7 +166,8 @@ function record_sensor_data($device, $all_sensors)
         $rrd_name = get_sensor_rrd_name($device, $sensor);
-        $rrd_def = RrdDefinition::make()->addDataset('sensor', 'GAUGE', -20000, 20000);
+
+        $rrd_def = RrdDefinition::make()->addDataset('sensor', 'GAUGE', -2000000000, 2000000000);
```
Can you just remove the limits here, thanks.  No reason to limit that I can think of.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/b636e947-8bca-4f12-af7c-57689b315908) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/b636e947-8bca-4f12-af7c-57689b315908) completed: **1 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/3e20332a-0d2d-43ed-95c6-148eb330e065) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/3e20332a-0d2d-43ed-95c6-148eb330e065) completed: **1 updated code elements**

laf commented on doc/Developing/os/Health-Information.md:
```diff
@@ -25,6 +25,10 @@ Currently we have support for the following health metrics along with the values
 | temperature                     | C                           |
 | voltage                         | V                           |
+| delay                           | s                           |
+| quality_factor                  | dB                          |
```
Any links to show that dB is the measurement of quality factor?

laf commented on doc/Developing/os/Health-Information.md:
```diff
@@ -25,6 +25,10 @@ Currently we have support for the following health metrics along with the values
 | voltage                         | V                           |
+| delay                           | s                           |
+| quality_factor                  | dB                          |
+| chromatic_disperision           | ps/nm                       |
```
Should this be ps/nm km?

laf commented on doc/Developing/os/Health-Information.md:
```diff
@@ -25,6 +25,10 @@ Currently we have support for the following health metrics along with the values
 | state                           | #                           |
 | temperature                     | C                           |
 | voltage                         | V                           |
+| delay                           | s                           |
```
Still not sure if ms is a better value to store

laf commented on html/includes/print-menubar.php:
```diff
@@ -392,6 +392,11 @@
+    'delay' => 'clock-o',
+    'chromatic_dispersion' => 'indent',
+    'ber' => 'sort-amount-desc',
+    'quality_factor' => 'arrows',
```
`area-chart` maybe?

bergroth commented:
On 2017-11-15 21:31, Neil Lathwood wrote:
>
> *@laf* commented on this pull request.
>
> Some queries around the measurement values for the new sensors.
>
> I think we can convert all of this to yaml with a little extra code. Would you
> be able to provide us with snmpwalk output for this device so we can simulate
> it and do the conversion?
>
sure what is the best way do deliver the file?

I had a couple of things I couldn't get to work with yaml.
1. To set description to  IF-MIB::ifDescr.index
2. to multiply two values to get BER: exponent and mantisa
3. the index for max an min has a different index compare to the interface index.

> --------------------------------------------------------------------------------
>
> In doc/Developing/os/Health-Information.md
> <https://github.com/librenms/librenms/pull/7714#discussion_r151241570>:
>
> > @@ -25,6 +25,10 @@ Currently we have support for the following health metrics along with the values
>  | state                           | #                           |
>  | temperature                     | C                           |
>  | voltage                         | V                           |
> +| delay                           | s                           |
> +| quality_factor                  | dB                          |
>
> Any links to show that dB is the measurement of quality factor?
>
https://www.juniper.net/documentation/en_US/junos/topics/concept/otn-signal-degrade-monitoring-understanding.html
http://mapyourtech.com/entries/general/what-is-q-factor-and-what-is-its-importance-

> --------------------------------------------------------------------------------
>
> In doc/Developing/os/Health-Information.md
> <https://github.com/librenms/librenms/pull/7714#discussion_r151241835>:
>
> > @@ -25,6 +25,10 @@ Currently we have support for the following health metrics along with the values
>  | state                           | #                           |
>  | temperature                     | C                           |
>  | voltage                         | V                           |
> +| delay                           | s                           |
> +| quality_factor                  | dB                          |
> +| chromatic_disperision           | ps/nm                       |
>
> Should this be ps/nm km?
>
ps/nm/km is the specification for the fiber. When you measure over the fiber
length it becomes ps/nm.
> --------------------------------------------------------------------------------
>
> In doc/Developing/os/Health-Information.md
> <https://github.com/librenms/librenms/pull/7714#discussion_r151241951>:
>
> > @@ -25,6 +25,10 @@ Currently we have support for the following health metrics along with the values
>  | state                           | #                           |
>  | temperature                     | C                           |
>  | voltage                         | V                           |
> +| delay                           | s                           |
>
> Still not sure if ms is a better value to store
>
I would be nice to have ms, us, ns and ps.
> --------------------------------------------------------------------------------
>
> In html/includes/print-menubar.php
> <https://github.com/librenms/librenms/pull/7714#discussion_r151242581>:
>
> > @@ -392,6 +392,11 @@
>      'pressure' => 'thermometer-empty',
>      'cooling' => 'thermometer-full',
>      'airflow' => 'angle-double-right',
> +    'delay' => 'clock-o',
> +    'chromatic_dispersion' => 'indent',
> +    'ber' => 'sort-amount-desc',
> +    'quality_factor' => 'arrows',
>
> |area-chart| maybe?
>
sure.
>
> —.
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/librenms/librenms/pull/7714#pullrequestreview-76918025>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASQkHrgz5Ir7DE4JSGpC68OHoNZgoeNfks5s20otgaJpZM4QbtIV>.
>

bergroth commented:
On 2017-11-15 21:31, Neil Lathwood wrote:
>
> *@laf* commented on this pull request.
>
> Some queries around the measurement values for the new sensors.
>
> I think we can convert all of this to yaml with a little extra code. Would you
> be able to provide us with snmpwalk output for this device so we can simulate
> it and do the conversion?
>

https://pastebin.com/Ac8c0Fke

laf commented:
> 1. To set description to  IF-MIB::ifDescr.index 2. to multiply two values to get BER: exponent and mantisa 3. the index for max an min has a different index compare to the interface index.

1. We can make that available by caching the data.

2. If you put that work into a function we can add support for passing a function in yaml.

3. This could be trickier, with the data we can see.

> juniper.net/documentation/en_US/junos/topics/concept/otn-signal-degrade-monitoring-understanding.html http://mapyourtech.com/entries/general/what-is-q-factor-and-what-is-its-importance-

Juniper link seems to suggest dBQ?

> I would be nice to have ms, us, ns and ps.

Yeah we can't do that :(

laf commented:
@bergroth we need the snmpwalk in this format: `snmpbulkwalk -OUneb -v2c -c COMMUNITY HOSTNAME .`

bergroth commented:
- Updated with -OUneb https://pastebin.com/Ac8c0Fke

- Q-factor according to the MIB are in unit dB
   jnxPMCurQ OBJECT-TYPE
        SYNTAX          Integer32
        UNITS           "0.1 dB"
        MAX-ACCESS      read-only
        STATUS          current
        DESCRIPTION
            "'Q' factor estimated at Rx Transceiver port "
        ::= { jnxOpticsPMCurrentEntry 5 }

and Cisco says the same
https://www.cisco.com/c/en/us/td/docs/optical/metroplanner/metroplanner_9_2_1/operations/guide/454mp921_opsguide/454mp921_interfacemodel.html


- Librenms handles the IS units nicely. Stored as seconds and displayed by RRD and Librenms as ps.
<img width="1259" alt="screen shot 2017-11-20 at 07 42 15" src="https://user-images.githubusercontent.com/19145758/33005548-908aece8-cdc6-11e7-956b-eed1bee50d72.png">

laf commented:
Thanks but can you provide the output as is from `snmpbulkwalk -OUneb -v2c -c COMMUNITY HOSTNAME .` - notice the trailing dot. Don't use any other OIDs. We need things like the interface data to match all of this up.

bergroth commented:
These are routers with full internet table, full snmpwalk going to be huge and
take a long time. Is the IF-MIB ok or do you need more? I might be able to get
one card up in the lab but it might take some time.

On 2017-11-20 16:49, Neil Lathwood wrote:
>
> Thanks but can you provide the output as is from |snmpbulkwalk -OUneb -v2c -c
> COMMUNITY HOSTNAME .| - notice the trailing dot. Don't use any other OIDs. We
> need things like the interface data to match all of this up.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/librenms/librenms/pull/7714#issuecomment-345736443>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASQkHllDbGWQOrE3C0rDfsMk4B0coe0Lks5s4Z9zgaJpZM4QbtIV>.
>

laf commented:
IF-MIB and Sensors might be ok. I need the data to replicate this PR basically.

I've got base junos data so that should be ok.

bergroth commented:
Included the if-mib as well

laf commented:
Got a link to the data?

bergroth commented:
same pastebin as before
https://pastebin.com/Ac8c0Fke

On 2017-11-22 22:36, Neil Lathwood wrote:
>
> Got a link to the data?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/librenms/librenms/pull/7714#issuecomment-346479817>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASQkHgKSs3qTiIImAvWfhWsaqshrb6hJks5s5JP3gaJpZM4QbtIV>.
>

laf commented:
Can you provide the data as one output in the format required. We don't need the data in both formats and split out like that,

laf commented:
This won't work without some back end changes I've made but this yaml for chromatic disperson works as expected:

```yaml
mib: JUNIPER-IFOPTICS-MIB:JNX-OPT-IF-EXT-MIB:IF-MIB
modules:
    sensors:
        pre-cache:
            data:
                - oid:
                    - ifDescr
        chromatic_dispersion:
            data:
                -
                    oid: jnxOpticsPMCurrentTable
                    value: jnxPMCurChromaticDispersion
                    num_oid: .1.3.6.1.4.1.2636.3.71.1.2.1.1.1.
                    entPhysicalIndex: '{{ $index }}'
                    entPhysicalIndex_measured: 'ports'
                    descr: '{{ $ifDescr }} CD'
                    index: 'jnxPMCurChromaticDispersion.{{ $index }}'
```

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/55e210be-498d-4d52-b8a9-71062c4d8604) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/55e210be-498d-4d52-b8a9-71062c4d8604) completed: **1 updated code elements**

laf commented:
@bergroth how you getting on with this?

bergroth commented:
I got all but the BER working with YAML.

BER

has two problems
1. it's two OIDs, exponent and mantisa that needs to calulated into a new value.
(using user_func)
2. the BER index is "ifindex.0". .0 needs to be stripped  for entPhysicalIndex.
Got it working with new code.

I can commit it tomorrow

<<M

On 2017-12-06 22:58, Neil Lathwood wrote:
>
> @bergroth <https://github.com/bergroth> how you getting on with this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/librenms/librenms/pull/7714#issuecomment-349788553>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASQkHhgpSHfJ66zc1oJDm3HBK-6bv3gpks5s9w4jgaJpZM4QbtIV>.
>

laf commented:
Have to keep ber in php for now then.

murrant commented:
@bergroth yeah, keep BER in php for now.

FYI, the wireless sensor code supports multiple oids and "aggregating" them.  But that code is very different from the sensor code, for now.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/044cde85-423c-4bb0-b03c-8b53d2484e28) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/044cde85-423c-4bb0-b03c-8b53d2484e28) completed: **1 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/47f251ed-590f-4598-8229-0692c72f5108) was created.

bergroth commented:
hope I got it right.

On 2017-12-10 21:28, Neil Lathwood wrote:
>
> *@laf* commented on this pull request.
>
> So nearly ok for me with two exceptions:
>
> 1.
>
>     You can't edit build.sql so please revert that chage.
>
> 2.
>
>     You've changed sensor_divisor to allow null and remove the default value,
>     please set allow null to false and the default divisor to 1.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/librenms/librenms/pull/7714#pullrequestreview-82361297>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASQkHtur2G1XF3-oG5FG0Shp7MaaMyd0ks5s_D7ggaJpZM4QbtIV>.
>

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/47f251ed-590f-4598-8229-0692c72f5108) completed: **86 new issues, 57 updated code elements**

murrant commented on build.sql:
```diff
@@ -41,7 +41,7 @@ CREATE TABLE IF NOT EXISTS `port_in_measurements` ( `port_id` int(11) NOT NULL,
 CREATE TABLE IF NOT EXISTS `processors` ( `processor_id` int(11) NOT NULL AUTO_INCREMENT, `entPhysicalIndex` int(11) NOT NULL, `hrDeviceIndex` int(11) DEFAULT NULL, `device_id` int(11) NOT NULL, `processor_oid` varchar(128) CHARACTER SET latin1 NOT NULL, `processor_index` varchar(32) CHARACTER SET latin1 NOT NULL, `processor_type` varchar(16) CHARACTER SET latin1 NOT NULL, `processor_usage` int(1 ...
 CREATE TABLE IF NOT EXISTS `pseudowires` ( `pseudowire_id` int(11) NOT NULL AUTO_INCREMENT, `interface_id` int(11) NOT NULL, `peer_device_id` int(11) NOT NULL, `peer_ldp_id` int(11) NOT NULL, `cpwVcID` int(11) NOT NULL, `cpwOid` int(11) NOT NULL, PRIMARY KEY (`pseudowire_id`) ) ENGINE=InnoDB  DEFAULT CHARSET=utf8;
-CREATE TABLE IF NOT EXISTS `sensors` ( `sensor_id` int(11) NOT NULL AUTO_INCREMENT, `sensor_class` varchar(64) CHARACTER SET latin1 NOT NULL, `device_id` int(11) NOT NULL DEFAULT '0', `poller_type` varchar(16) COLLATE utf8_unicode_ci NOT NULL DEFAULT 'snmp', `sensor_oid` varchar(64) CHARACTER SET latin1 NOT NULL, `sensor_index` varchar(10) COLLATE utf8_unicode_ci NOT NULL, `sensor_type` varchar(2 ...
+CREATE TABLE IF NOT EXISTS `sensors` ( `sensor_id` int(11) NOT NULL AUTO_INCREMENT, `sensor_class` varchar(64) CHARACTER SET latin1 NOT NULL, `device_id` int(11) NOT NULL DEFAULT '0', `poller_type` varchar(16) COLLATE utf8_unicode_ci NOT NULL DEFAULT 'snmp', `sensor_oid` varchar(64) CHARACTER SET latin1 NOT NULL, `sensor_index` varchar(10) COLLATE utf8_unicode_ci NOT NULL, `sensor_type` varchar(2 ...
```
You cannot change history.  Please add this to your new schema sql file.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/1d05fda6-65ae-48df-b85e-409c5d25de82) was created.

bergroth commented:
I reverted that file, @laf said that I couldn't change that file. corrected
db_schema.yaml

On 2017-12-11 16:55, Tony Murray wrote:
>
> *@murrant* commented on this pull request.
>
> --------------------------------------------------------------------------------
>
> In build.sql
> <https://github.com/librenms/librenms/pull/7714#discussion_r156113313>:
>
> > @@ -41,7 +41,7 @@ CREATE TABLE IF NOT EXISTS `port_in_measurements` ( `port_id` int(11) NOT NULL,
>  CREATE TABLE IF NOT EXISTS `port_out_measurements` ( `port_id` int(11) NOT NULL, `timestamp` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP, `counter` bigint(11) NOT NULL, `delta` bigint(11) NOT NULL, KEY `port_id` (`port_id`) ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
>  CREATE TABLE IF NOT EXISTS `processors` ( `processor_id` int(11) NOT NULL AUTO_INCREMENT, `entPhysicalIndex` int(11) NOT NULL, `hrDeviceIndex` int(11) DEFAULT NULL, `device_id` int(11) NOT NULL, `processor_oid` varchar(128) CHARACTER SET latin1 NOT NULL, `processor_index` varchar(32) CHARACTER SET latin1 NOT NULL, `processor_type` varchar(16) CHARACTER SET latin1 NOT NULL, `processor_usage` int(11) NOT NULL, `processor_descr` varchar(64) CHARACTER SET latin1 NOT NULL, `processor_precision` int(11) NOT NULL DEFAULT '1', PRIMARY KEY (`processor_id`), KEY `device_id` (`device_id`) ) ENGINE=InnoDB  DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
>  CREATE TABLE IF NOT EXISTS `pseudowires` ( `pseudowire_id` int(11) NOT NULL AUTO_INCREMENT, `interface_id` int(11) NOT NULL, `peer_device_id` int(11) NOT NULL, `peer_ldp_id` int(11) NOT NULL, `cpwVcID` int(11) NOT NULL, `cpwOid` int(11) NOT NULL, PRIMARY KEY (`pseudowire_id`) ) ENGINE=InnoDB  DEFAULT CHARSET=utf8;
> -CREATE TABLE IF NOT EXISTS `sensors` ( `sensor_id` int(11) NOT NULL AUTO_INCREMENT, `sensor_class` varchar(64) CHARACTER SET latin1 NOT NULL, `device_id` int(11) NOT NULL DEFAULT '0', `poller_type` varchar(16) COLLATE utf8_unicode_ci NOT NULL DEFAULT 'snmp', `sensor_oid` varchar(64) CHARACTER SET latin1 NOT NULL, `sensor_index` varchar(10) COLLATE utf8_unicode_ci NOT NULL, `sensor_type` varchar(255) CHARACTER SET latin1 NOT NULL, `sensor_descr` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL, `sensor_divisor` int(11) NOT NULL DEFAULT '1', `sensor_multiplier` int(11) NOT NULL DEFAULT '1', `sensor_current` float DEFAULT NULL, `sensor_limit` float DEFAULT NULL, `sensor_limit_warn` float DEFAULT NULL, `sensor_limit_low` float DEFAULT NULL, `sensor_limit_low_warn` float DEFAULT NULL, `entPhysicalIndex` varchar(16) CHARACTER SET latin1 DEFAULT NULL, `entPhysicalIndex_measured` varchar(16) CHARACTER SET latin1 DEFAULT NULL, PRIMARY KEY (`sensor_id`), KEY `sensor_host` (`device_id`), KEY `sensor_class` (`sensor_class`), KEY `sensor_type` (`sensor_type`) ) ENGINE=InnoDB  DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
> +CREATE TABLE IF NOT EXISTS `sensors` ( `sensor_id` int(11) NOT NULL AUTO_INCREMENT, `sensor_class` varchar(64) CHARACTER SET latin1 NOT NULL, `device_id` int(11) NOT NULL DEFAULT '0', `poller_type` varchar(16) COLLATE utf8_unicode_ci NOT NULL DEFAULT 'snmp', `sensor_oid` varchar(64) CHARACTER SET latin1 NOT NULL, `sensor_index` varchar(10) COLLATE utf8_unicode_ci NOT NULL, `sensor_type` varchar(255) CHARACTER SET latin1 NOT NULL, `sensor_descr` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL, `sensor_divisor` bigint(20) NOT NULL DEFAULT '1', `sensor_multiplier` int(11) NOT NULL DEFAULT '1', `sensor_current` float DEFAULT NULL, `sensor_limit` float DEFAULT NULL, `sensor_limit_warn` float DEFAULT NULL, `sensor_limit_low` float DEFAULT NULL, `sensor_limit_low_warn` float DEFAULT NULL, `entPhysicalIndex` varchar(16) CHARACTER SET latin1 DEFAULT NULL, `entPhysicalIndex_measured` varchar(16) CHARACTER SET latin1 DEFAULT NULL, PRIMARY KEY (`sensor_id`), KEY `sensor_host` (`device_id`), KEY `sensor_class` (`sensor_class`), KEY `sensor_type` (`sensor_type`) ) ENGINE=InnoDB  DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
>
> You cannot change history. Please add this to your new schema sql file.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/librenms/librenms/pull/7714#pullrequestreview-82538440>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASQkHpTkvi00OMir_XNgUK7S9yqUyAwoks5s_VBggaJpZM4QbtIV>.
>

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/1d05fda6-65ae-48df-b85e-409c5d25de82) completed: **No new issues**

laf commented on includes/discovery/sensors/current/junos.inc.php:
```diff
@@ -17,14 +17,14 @@
         $entPhysicalIndex = $index;
         $entPhysicalIndex_measured = 'ports';
-        discover_sensor($valid['sensor'], 'current', $device, $oid, 'rx-' . $index, 'junos', $descr, $divisor, $multiplier, $limit_low, $warn_limit_low, $warn_limit, $limit, $current, 'snmp', $entPhysicalIndex, $entPhysicalIndex_measured);
+        discover_sensor($valid['sensor'], 'current', $device, $oid, 'tx-' . $index, 'junos', $descr, $divisor, $multiplier, $limit_low, $warn_limit_low, $warn_limit, $limit, $current, 'snmp', $entPhysicalIndex, $entPhysicalIndex_measured);
```
Annoyingly, we can't change this from rx-. It will mean data loss for these sensors :(

laf commented on sql-schema/223.sql:
```diff
@@ -0,0 +1 @@
+ALTER TABLE `sensors` MODIFY `sensor_divisor` BIGINT(20) NOT NULL DEFAULT '1';
```
we have the potential for another schema change for sensors here. I'd like to get that into this PR so we only do one overall change to the table. Just waiting on confirmation that it fixes another issue.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/57ef6b62-4873-45c4-a49e-977d90f743fa) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/57ef6b62-4873-45c4-a49e-977d90f743fa) completed: **1 updated code elements**
